packages/clay-core/src/table/Table.tsx

```diff
@@ -48,6 +48,13 @@ export type Props = {
 	 */
 	onExpandedChange?: (keys: Set<React.Key>) => void;
 
+	/**
+	 * When a tree is very large, loading items (nodes) asynchronously is preferred to
+	 * decrease the initial payload and memory space. The callback is called every time
+	 * the item is a leaf node of the tree.
```

Comment:
So it's called when the item is "currently" a leaf, because the leaf might have children that need to be loaded? Is that right?

Reply:
Yes, this is the idea. We could implement paginated loading for non-leaf elements as well, as exists in TreeView, but I think it's best to wait for use cases to appear for this because it involves UI changes as well, such as where to add the button to load more, etc...